File: reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int rev = 0;
        while(x != 0){
            int lastDigit = x % 10;
            x = x / 10;
            if(rev > Integer.MAX_VALUE/10 || (rev == Integer.MAX_VALUE/10 && lastDigit > 7)) return 0;
            if(rev < Integer.MIN_VALUE/10 || (rev == Integer.MIN_VALUE/10 && lastDigit < -8)) return 0;
            rev = (rev * 10) + lastDigit;
        }
        return rev;
    }
}

File: 7-reverse-integer/reverse-integer.ts
function reverse(x: number): number {
    let xCopy = x;
    let rev = 0;
    x = Math.abs(x);

    while (x > 0) {
        let last_digit = x % 10;
        rev = (rev * 10) + last_digit;
        x = Math.floor(x / 10);
    }

    let limit = Math.pow(2, 31);
    if (rev < -limit || rev > limit) return 0;

    return (xCopy < 0) ? -rev : rev;
};